cmd/kubelet/app/server.go

```diff
@@ -457,7 +457,7 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.KubeletDeps) (err error) {
 
 	if kubeDeps.CAdvisorInterface == nil {
 		kubeDeps.CAdvisorInterface, err = cadvisor.New(uint(s.CAdvisorPort), s.ContainerRuntime, s.RootDirectory)
-		if err != nil {
+		if err != nil && goruntime.GOOS != "darwin" {
```

Comment:
we don't typically do GOOS checks to ignore errors like this. A better option would be to make a platform-specific stubbed implementation of the relevant interface, and use that when GOOS positively identifies a particular platform